### PR TITLE
355 improve patient creation modal

### LINKED ISSUE #355: Imrove patient registration dialog
Add:

- [x] Sex
- [x] Date of birth / Age as an alternative to DOB, e.g. if unknown (related to #385)
- [x] Patient id (Patient code)

### PR DESCRIPTION
**Pull request by clemens-msupply**
Adds id (code), gender and dob to the patient search. Extend api to support the new fields.

Open questions:
[#407](https://github.com/openmsupply/open-msupply/issues/407)
[#408](https://github.com/openmsupply/open-msupply/issues/408)

Closes #355

### CODE REVIEW

File: client/packages/system/src/Patient/CreatePatientModal/PatientFormTab.tsx [hunk 1]
```diff
@@ -1,20 +1,45 @@
...
+          Input={
+            <BufferedTextInput
+              size="small"
+              sx={{ width: 250 }}
+              value={patient?.code ?? ''}
+              onChange={event => updatePatient({ code: event.target.value })}
+              autoFocus
+            />
+          }
+        />
```
Comment (mark-prins) [inline client/packages/system/src/Patient/CreatePatientModal/PatientFormTab.tsx:42]: A suggestion to reduce the duplication here..
Outside of the component add this:
```
interface PatientInputRowProps {
  autoFocus?: boolean;
  field: keyof CreateNewPatient;
  label: string;
  patient?: CreateNewPatient;
  updatePatient: (patient: Partial<CreateNewPatient>) => void;
}

const PatientInputRow: FC<PatientInputRowProps> = ({
  autoFocus,
  label,
  field,
  patient,
  updatePatient,
}) => (
  <InputWithLabelRow
    label={label}
    Input={
      <BufferedTextInput
        size="small"
        sx={{ width: 250 }}
        value={patient?.[field] ?? ''}
        onChange={event => updatePatient({ [field]: event.target.value })}
        autoFocus={autoFocus}
      />
    }
  />
);
```

and then
```
<PatientInputRow
          label={t('label.id')}
          field="id"
          patient={patient}
          updatePatient={updatePatient}
        />
        <PatientInputRow
          label={t('label.first-name')}
          field="firstName"
          patient={patient}
          updatePatient={updatePatient}
          autoFocus
        />
        <PatientInputRow
          label={t('label.last-name')}
          field="lastName"
          patient={patient}
          updatePatient={updatePatient}
        />
```

**Review by mark-prins — APPROVED** [review]
👍 that's very nice!

File: client/packages/system/src/Patient/CreatePatientModal/PatientFormTab.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/system/src/Patient/CreatePatientModal/PatientFormTab.tsx:42]: It only makes it a little bit shorter and slightly more complex and then there are the BaseDatePickerInput and Select input rows which don't fit this pattern. Do you mind leaving it as is?

Comment (mark-prins) [reply client/packages/system/src/Patient/CreatePatientModal/PatientFormTab.tsx:42]: no, that's fine to leave as is

**Pull request merged**